pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -115,10 +116,19 @@ func (c TargetConfigController) sync(ctx context.Context, syncContext factory.Sy
 	}
 
 	cmString := string(c.manifest)
+
+	virtualCenterIPs := sets.StringKeySet(cfg.VirtualCenter)
+
+	if len(virtualCenterIPs) > 1 || len(virtualCenterIPs) == 0 {
+		return fmt.Errorf("cloud config %s/%s must define a single VirtualCenter", cloudConfigNamespace, cloudConfig.Name)
+	}
+
+	virtualCenterIP := virtualCenterIPs.List()[0]
```

Comment:
@gnufied @bertinatto Do you see a way for us to actually support multiple datacenters? Or is this the best option we have for now (throw a clear error)?

Reply:
The restriction here is actually a single vCenter.  With multi-zone support, it is very possible a customer may have multiple datacenters.

Reply:
and that could probably just be refactored to `if len(virtualCenterIPs) != 1` . :facepalm:

Reply:
So we traditionally picked this from workspace and now we are changing it to vcenter section. Could we break anyone? Such as UPI installs or people who have modified their cloud-configs? Shouldn't workspace datacenter and `VirtualCenter` datacenter be the same?